.github/workflows/test-e2e-runtime-watcher.yml

```diff
@@ -0,0 +1,156 @@
+name: TestSuite E2E
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+jobs:
+  wait-for-img:
+    name: "Wait for Image Build"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: autotelic/action-wait-for-status-check@v1
+        id: wait-for-build
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          # Context for which we should look for the matching status
+          statusName: ${{ (github.event_name == 'pull_request') && 'pull-runtime-watcher-img-build' || 'main-runtime-watcher-build' }}
+          timeoutSeconds: 900
+          intervalSeconds: 10
+      - name: Exit If Failing Build Requirement
+        if: steps.wait-for-build.outputs.state != 'success'
+        run: |
+          echo "Image build did not succeed, skipping E2E Test!"
+          exit 1
+
+  e2e-integration:
+    name: "E2E"
+    needs: [wait-for-img]
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./runtime-watcher
+    timeout-minutes: 20
+    env:
+      K3D_VERSION: v5.4.7
```

Comment:
Good idea! Since the feature is beta and requires some alignment with lm, let's keep it for a separate issue.
I have collected it here: https://github.com/kyma-project/lifecycle-manager/issues/1224